packages/fixie/src/main.ts

```diff
@@ -179,14 +179,36 @@ sources
   .description('Add a source to a corpus.')
   .option('--max-documents <number>', 'Maximum number of documents to crawl')
   .option('--max-depth <number>', 'Maximum depth to crawl')
+  .option('--include-patterns <pattern...>', 'URL patterns to include in the crawl')
+  .option('--exclude-patterns <pattern...>', 'URL patterns to exclude from the crawl')
   .action(
     async (
       corpusId: string,
       startUrls: string[],
-      { maxDocuments, maxDepth }: { maxDocuments?: number; maxDepth?: number }
+      {
+        maxDocuments,
+        maxDepth,
+        includePatterns,
+        excludePatterns,
+      }: { maxDocuments?: number; maxDepth?: number; includePatterns?: string[]; excludePatterns: string[] }
     ) => {
+      if (!includePatterns) {
+        term.yellow('Warning: ')(
+          'No --include-patterns specfied. This is equivalent to only crawling the URLs specified as startUrls.\n'
+        );
+        term.yellow('Warning: ')('Use ').red("--include-patterns '*'")(
```

Comment:
Is there any scenario where this is actually a reasonable thing to do?

Reply:
Sure, if you want to rely on max-depth and max-documents and crawl anything you find from a start URL. If you have a better proposed default, I'm all for it, but I don't know what it would be :-)

Reply:
My concern is just if the suggestion is overwhelmingly likely to lead to a slow crawl that then fails out...it's worse than not suggesting anything, or simply letting the user consult docs to understand how to use `--include-patterns`.

But if hitting the max is "fine" (i.e. neither users nor Fixie is unhappy) then that's not a concern.